applications/ContactStructuralMechanicsApplication/custom_utilities/contact_utilities.cpp

```diff
@@ -211,6 +211,25 @@ bool ContactUtilities::CheckActivity(
 /***********************************************************************************/
 /***********************************************************************************/
 
+bool ContactUtilities::CheckModelPartHasRotationDoF(ModelPart& rModelPart)
+{
+    auto& r_nodes_array = rModelPart.Nodes();
+    for(auto& r_node : r_nodes_array) {
+        const auto& r_dofs = r_node.GetDofs();
+        for (auto it_dof = r_dofs.begin(); it_dof != r_dofs.end(); ++it_dof) {
+            const auto& r_variable = (**it_dof).GetVariable();
+            if (r_variable == ROTATION_X || r_variable == ROTATION_Y || r_variable == ROTATION_Z) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
```

Comment:
wouldn't it be easier to check the ModelPart directly (=> also name is kinda misleading)?
=> in most cases the ModelPart has the same Variables as the Nodes

if you want to REALLY check then you would need to ask the Elements for their `GetDofList`

Reply:
Yes, I thought about that, but this was simpler...